components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediator.java

```diff
@@ -359,17 +364,19 @@ private void processResponseMessage(MessageContext synCtx, ConfigurationContext
         CachableResponse response = (CachableResponse) synCtx.getProperty(CachingConstants.CACHED_OBJECT);
 
         if (response != null) {
-            boolean toCache;
+            boolean toCache = true;
```

Comment:
Explicit declaration of toCache is redundant. if it's false, we have to update the variable to true, if it matches the pattern, and we can avoid else block.

Reply:
Yes it is possible to avoid the else statement. I have changed it. This declaration has to be there because if it is not the http protocol type without checking the pattern it should cache all.